tests/variable_test.py

```diff
@@ -0,0 +1,353 @@
+import defcon
+import fontTools
+import pytest
+from fontTools.designspaceLib import (
+    AxisDescriptor,
+    DesignSpaceDocument,
+    InstanceDescriptor,
+    SourceDescriptor,
+)
+
+from ufo2ft import (
+    compileInterpolatableOTFsFromDS,
+    compileInterpolatableTTFs,
+    compileInterpolatableTTFsFromDS,
+    compileVariableCFF2,
+    compileVariableTTF,
+)
+
+
+def make_test_font(family_name, style_name, outline_width=100):
+    """Make a test UFO font with editable family and style names, and
+    with one glyph whose outline width is parametrized"""
+
+    f = defcon.Font()
+    f.info.familyName = family_name
+    f.info.openTypeNamePreferredFamilyName = family_name
+    f.info.styleName = style_name
+    f.info.openTypeNamePreferredSubfamilyName = style_name
+    f.info.postscriptFontName = f"{family_name.replace(' ', '')}-{style_name}"
+    f.info.ascender = 750
+    f.info.capHeight = 700
+    f.info.descender = -250
+    f.info.xHeight = 500
+    f.info.unitsPerEm = 1000
+
+    glyph = f.newGlyph("I")
+    glyph.unicode = 0x0049
+    pen = glyph.getPen()
+    margin = 100
+    height = 750
+    pen.moveTo((margin, 0))
+    pen.lineTo((margin + outline_width, 0))
+    pen.lineTo((margin + outline_width, height))
+    pen.lineTo((margin, height))
+    pen.closePath()
+    glyph.width = outline_width + (margin * 2)
+
+    return f
+
+
+def regular_font():
+    return make_test_font("Test Font Names", "Regular", outline_width=100)
+
+
+def heavy_font():
+    return make_test_font("Test Font Names", "Heavy", outline_width=500)
+
+
+def make_instance(name, style_name, location):
+    instance = InstanceDescriptor()
+    instance.name = f"{name}.{style_name}"
+    instance.styleName = style_name
+    instance.location = location
+    return instance
+
+
+def make_designspace(
+    source_family_name_override=None, source_style_name_override=None, instances=()
+):
+    doc = DesignSpaceDocument()
+    a1 = AxisDescriptor()
+    a1.minimum = 400
+    a1.maximum = 800
+    a1.default = 400
+    a1.name = "weight"
+    a1.tag = "wght"
+    doc.addAxis(a1)
+
+    s1 = SourceDescriptor()
+    s1.name = "master.TestFontNames.Regular.400"
+    s1.font = regular_font()
+    s1.copyLib = True
+    s1.copyInfo = True
+    s1.copyFeatures = True
+    s1.location = dict(weight=400)
+    if source_family_name_override is not None:
+        s1.familyName = source_family_name_override
+    if source_style_name_override is not None:
+        s1.styleName = source_style_name_override
+    doc.addSource(s1)
+
+    s2 = SourceDescriptor()
+    s2.name = "master.TestFontNames.Heavy.800"
+    s2.font = heavy_font()
+    s2.location = dict(weight=800)
+    doc.addSource(s2)
+
+    if instances:
+        for i in instances:
+            doc.addInstance(i)
+
+    return doc
+
+
+def make_designspace_plain(
```

Comment:
what's the use of `make_designspace_plain` function? it does exactly the same thing as `make_designspace` function, you can simply remove the former and replace it with calls to the latter

Reply:
ah, I see, it populates `instances`.